### DISCUSSION TITLE
Kong integration

### DISCUSSION THREAD

**Discussion by Nono31** [authelia/authelia#4883] [Ideas]
### Description

As a kong user, I'm looking for authelia integration on kong API gateway.
I'm looking for a kong plugin compatible with authelia

### Use Case

As a kong gateway user I want to be able to authenticate through authelia

### Documentation

[Kong](https://docs.konghq.com/)

### Pre-Submission Checklist

- [X] I agree to follow the [Code of Conduct](http://www.authelia.com/code-of-conduct)
- [X] I have checked for related issues and checked the documentation

**Comment by james-d-elliott** [comment]
Kong Gateway [supports OpenID Connect 1.0](https://docs.konghq.com/gateway/latest/kong-plugins/authentication/oidc/) and we [support OpenID Connect 1.0](https://www.authelia.com/integration/openid-connect/introduction/), and Kong Gateway is built on top of [NGINX](https://nginx.org/en/) and [we support an NGINX authz](https://www.authelia.com/integration/proxies/nginx/) via the [ngx_http_auth_request_module](https://nginx.org/en/docs/http/ngx_http_auth_request_module.html). There [may already be plugins](https://discuss.konghq.com/t/kong-plugin-for-forward-authentication/7838) which support this, or see [this topic](https://github.com/Kong/kong/discussions/7779).

Has any of these methods been investigated?

We support several widely adopted authz methods including ExtAuthz, Forward Auth, nginx auth request. We are also very open to working with developers to facilitate them developing such integrations on their end as we did with Caddy and Skipper. 

I feel like this belongs as a discussion for users to try and find methods that already exist and are likely to work. But I will give you a chance to respond before converting it, if I do.

**Comment by Nono31** [comment]
About the plugins, first wada-ama/kong-plugin-opa doesnt work, it just keeps kong restart with an error, the second one TravelNest/kong-authorization-opa is old, very old, "Created to work with Kong v0.13.x"
And about the topic; "Kong does not currently have this functionality built-in"

I keep looking at opa plugins, or other solutions, but I'm pretty much struggled

**Comment by james-d-elliott** [comment]
I've converted this to a discussion where people can vote on this idea for us to take this up as a feature for design and/or provide their individual attempts at getting this working.

**Comment by james-d-elliott** [comment]
https://github.com/ascho/kong-auth-request
